command/agent/cache/lease_cache.go

```diff
@@ -83,6 +85,13 @@ type LeaseCache struct {
 
 	// inflightCache keeps track of inflight requests
 	inflightCache *gocache.Cache
+
+	// ps is the persistent storage for tokens and leases
+	ps *cacheboltdb.BoltStorage
+
+	// shuttingDown is used to determine if cache needs to be evicted or not
+	// when the context is cancelled
+	shuttingDown bool
```

Comment:
I'd suggest using an [atomic bool](https://pkg.go.dev/go.uber.org/atomic#Bool) value since this can be modified by rungroups.

Reply:
Changed in https://github.com/hashicorp/vault/pull/10938/commits/0dd3ae4cdddefe9fa567b29be9c25fbd3a51fb93